File: 240718/treeset 기본/treeset-basic.js
const sortedSet = require("collections/sorted-set");
const fs = require('fs');

const commands = fs.readFileSync(0).toString().trim().split("\n")
.slice(1).map(i => i.trim().split(" "));

let ss = new sortedSet();

commands.forEach(com => {
    switch(com[0]){
        case "add":
            ss.add(Number(com[1]));
            break;
        case "remove":
            ss.delete(Number(com[1]));
            break;
        case "find":
            console.log(ss.has(Number(com[1])));
            break;
        case "lower_bound":
            if (ss.findLeastGreaterThanOrEqual(Number(com[1])))
                console.log(ss.findLeastGreaterThanOrEqual(Number(com[1])).value)
            else console.log("None");
            break;
        case "upper_bound":
            if (ss.findLeastGreaterThan(Number(com[1])))
                console.log(ss.findLeastGreaterThan(Number(com[1])).value)
            else console.log("None");
            break;
        case "smallest":
            if (ss.size > 0) console.log(ss.findLeast().value);
            else console.log("None");
            break;
        case "largest":
            if (ss.size > 0) console.log(ss.findGreatest().value);
            else console.log("None");
            break;
    }
})

File: 240718/treeset 기본/treeset-basic.kt
import java.io.*
import java.util.*
import kotlin.collections.*

val set = TreeSet<Int>()

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val num = br.readLine().toInt()
    for (i in 0 until num) {
        val command =
            br
                .readLine()
                .trim()
                .split(" ")
                .toTypedArray()
        when (command[0]) {
            "add" -> set.add(command[1].toInt())
            "remove" -> set.remove(command[1].toInt())
            "find" -> println(set.contains(command[1].toInt()))
            "lower_bound" ->
                println(
                    set.contains(command[1].toInt()).let {
                        if (it) {
                            command[1].toInt()
                        } else {
                            set.higher(command[1].toInt())
                        }
                    } ?: "None"
                )
            "upper_bound" -> println(set.higher(command[1].toInt()) ?: "None")
            "largest" -> println(if (set.size > 0) set.last() else "None")
            "smallest" -> println(if (set.size > 0) set.first() else "None")
        }
    }
}